internal/cortex/querier/queryrange/queryrange.proto

```diff
@@ -62,6 +63,7 @@ message PrometheusInstantQueryData {
   string ResultType = 1 [(gogoproto.jsontag) = "resultType"];
   PrometheusInstantQueryResult Result = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "result"];
   PrometheusResponseStats stats = 3 [(gogoproto.jsontag) = "stats,omitempty"];
+  Explanation Explanation = 4 [(gogoproto.jsontag) = "explanation"];
```

Comment:
Umm so explanation is part of the instant query and range query responses?

Reply:
Yes, this field will be expanded in the future with information such as CPU time taken by each operator so it needs to be part of this API. 
I thought about moving all of this under the Thanos codec but that will mean some copy/pasted code. Do you think that approach would be better? Or maybe you have some other suggestions or ideas? 🤔

Reply:
I am wondering if we should move it to `stats` and extend there? At current stage a separate field is fine.

> I thought about moving all of this under the Thanos codec but that will mean some copy/pasted code.

If we move the logic under codec, we still need additional fields in the response to be decoded into explanation, right? What's the difference

Reply:
Perhaps in the future, my concern is that `stats` is currently reserved by Prometheus and ideally we should avoid any potential clashes in the future.

My point is that `PrometheusInstantQueryData` with this change doesn't reflect the reality properly because it is called "Prometheus" even though it would contain Thanos-specific fields. Perhaps it's not a problem since we have a copy of the Cortex code either way? An alternative would be to copy this whole struct, rename it to "ThanosInstantQueryData", and copy all of the encoding/decoding logic together with handling of the new field.